### ISSUE TITLE
您好，我能这么写用来检测友军吗，不知道为什么运行失败，大佬求帮

### ISSUE THREAD

**Issue by GrayTempest-400** [mrathena/python.yolo.apex.autoaim.helper#4]
#敌我识别                                                                                                                  
                                                                                                                       
 try:                                                                                                                  
     img = cv2.imread(img)                                                                                             
                                                                                                                       
                                                                                                                       
 # 将图像转换为RGB格式                                                                                                         
     img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)                                                                        
                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                            # 定义蓝色和绿色的HSV范围                                                                                                       
     lower_blue = np.array([110, 50, 50])                                                                              
     upper_blue = np.array([130, 255, 255])                                                                            
     lower_green = np.array([50, 50, 50])                                                                              
     upper_green = np.array([70, 255, 255])                                                                            
                                                                                                                       
 # 获取box的位置和大小                                                                                                         
     x1 = target[3][0],y1 = target[3][1],x2 = target[3][2],y2 = target[3][3]                                           
     cx = (x1 + x2) / 2                                                                                                
     cy = (y1 + y2) / 2                                                                                                
     w = x2 - x1                                                                                                       
     h = y2 - y1                                                                                                       
     target = torch.stack((cx, cy, w, h), axis=-1)                                                                     
     print(f'{x1},{y1},{x2},{y2}')                                                                                     
                                                                                                                       
     x, y, w, h = (cx, cy, w, h)   #这段可能还有问题                                                                           
                                                                                                                       
                                                                                                                       
     roi = img[y - 100:y, x:x + w]      # 获取box上方100个y内的区域                                                             
                                                                                                                       
                 # 将区域转换为HSV颜色空间                                                                                       
     hsv_roi = cv2.cvtColor(roi, cv2.COLOR_BGR2HSV)                                                                    
                                                                                                                       
     # 创建掩膜                                                                                                            
     mask_roi_blue = cv2.inRange(hsv_roi, lower_blue, upper_blue)                                                      
     mask_roi_green = cv2.inRange(hsv_roi, lower_green, upper_green)                                                   
 except:                                                                                                               
   print("no")                                                                                                         
 # 如果检测到蓝色或绿色                                                                                                          
 if (cv2.countNonZero(mask_roi_blue) == 0) and (cv2.countNonZero(mask_roi_green) == 0)and data[lock] and target :      
     index, clazz, conf, sc, gc, sr, gr = target                                                                       
     if inner(sc):                                                                                                     
         cx, cy = data[center]                                                                                         
         sx, sy = sc                             #移动十字准线以跟随目标对象                                                        
         x = sx - cx                                                                                                   
         y = sy - cy                                                                                                   
         if data[pidc]:                                                                                                
             px = -int(pidx(x))                                                                                        
             move(px, y)                                                                                               
         else:                                                                                                         
             ax = int(x * data[ads])                                                                                   
             ay = int(y * data[ads])                                                                                   
             move(ax, ay)
